superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/index.tsx

```diff
@@ -289,14 +289,14 @@ const granularity_sqla: SharedControlConfig<'SelectControl', ColumnMeta> = {
   mapStateToProps: state => {
     const props: Partial<SelectControlConfig<ColumnMeta | QueryColumn>> = {};
     const { datasource } = state;
-    if (datasource?.columns[0]?.hasOwnProperty('main_dttm_col')) {
+    if (datasource?.hasOwnProperty('main_dttm_col')) {
```

Comment:
The `main_dttm_col` property exists on the datasource object as opposed the associated columns.